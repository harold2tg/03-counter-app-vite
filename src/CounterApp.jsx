import { useState } from 'react';
import PropTypes from 'prop-types'

export const CounterApp = ({value}) =>{
    
    const [ counter, setCounter ] = useState(0)

    function handleAdd(){
        setCounter( (counter)=> counter + 1)
    }

    function handleSubstract(){
        setCounter( (counter)=> counter - 1)
    }

    function handleReset(){
    setCounter( (counter)=> counter=0)
    }

    return(
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleSubstract }>-1</button>
            <button onClick={ handleReset }>Reset</button>
            
        </>
    );


}

CounterApp.propTypes = {
    value:PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value:0
}