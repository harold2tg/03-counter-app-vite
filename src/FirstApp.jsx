
// import { Fragment } from "react"

const nombre = "Harold"

const persona={
    nombre:'Harold',
    profesion:'Ingeniero'
}

const animal={
    nombre:'pato',
    profesion:'joder'
}

const devolveralgo = ()=> {
    return animal
}

export const FirstApp = ({title='Hola soy Harold'}) => {

  return (
    // <Fragment>
    <>
    <h1>First App</h1>
    <p>Hola soy {nombre}</p>
    <h5>{1 + 1 }</h5>

    <h4>Como mostrar un objeto:</h4>
    <p>{JSON.stringify(persona)}</p>

    <h4>Nombre: {persona.nombre}</h4>
    <h4>Profesi√≥n: {persona.profesion}</h4>

    <h1>Mostrar animales</h1>
    <p>{animal.nombre}</p>
    <p>{animal.profesion}</p>


    <h2> 1props</h2>
    <p>soy {title}</p>

    </>
    // </Fragment>
  )
}
