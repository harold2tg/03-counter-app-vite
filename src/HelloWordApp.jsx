
// function App(){
//     return <h1>Hola mundo !!!</h1>;
// }

// export default App;


export const HelloWordApp = () => {
  return (
    <h1>Hello Word App</h1>
  )
}
