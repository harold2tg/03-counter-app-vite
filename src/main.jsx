import React from 'react';
import ReactDOM from 'react-dom/client';
// import { App } from './HelloWordApp';
// import App from './HelloWordApp';
import {HelloWordApp} from './HelloWordApp';
import {FirstApp} from './FirstApp';
import './styles.css';



ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWordApp/>
        <FirstApp/>
    </React.StrictMode>
)