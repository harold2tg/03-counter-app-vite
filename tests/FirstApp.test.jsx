import {render} from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Probando componente FirstApp', () => { 
    test('Debe hacer match con snatshot', () => { 

        const titulo = 'Hola soy Harold'
        const subtitulo = 'Soy ingeniero de sistemas'
        const nombre = 'Hola soy Hacreado por isaias'

        const {container } = render(<FirstApp title={titulo}/>)
        expect (container).toMatchSnapshot();

     })
     test('debe mostrar el titulo en h1', () => { 
        const titulo = 'Hola soy Harold'
        const {container, getByText } = render(<FirstApp title={titulo}/>)
        expect (getByText(titulo)).toBeTruthy();
        const h1 = container.querySelector('h1')
        console.log(h1.innerHTML)
        expect(h1.innerHTML).toContain(titulo)

      })
 })