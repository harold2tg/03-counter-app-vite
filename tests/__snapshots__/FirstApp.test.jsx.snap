// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente FirstApp Debe hacer match con snatshot 1`] = `
<div>
  <h1>
    Hola soy Harold
  </h1>
  <h2>
    Nuevo sub-titulo
  </h2>
  <p>
    sin nombre
  </p>
</div>
`;
